### PR TITLE
[13.0][ADD] account_move_line_sale_info

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
Supersedes #1032

This module will add the sale order line to journal items. This is useful for reconciling the Stock interim account (Output) when using automated inventory valuation. For keeping that account clean users can reconcile it at sale order level

@ForgeFlow

**EDIT: Added a hook that allows to modify standard criteria for why the anglosaxon entries are posted, and the values on it**

### CODE REVIEW

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

File: stock_account_prepare_anglo_saxon_out_lines_hook/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2020 ForgeFlow S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Account Move Anglo saxon Hook",
```
Comment (MiquelRForgeFlow) [inline stock_account_prepare_anglo_saxon_out_lines_hook/__manifest__.py]: needs update

Comment (AaronHForgeFlow) [reply stock_account_prepare_anglo_saxon_out_lines_hook/__manifest__.py]: Done, thanks

File: account_move_line_sale_info/hooks.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+def post_init_hook(cr, registry):
+
+    """ INIT sale references in acount move line """
+    # FOR stock moves
+    cr.execute(
+        """
+        update account_move_line aml set sale_line_id = sm.sale_line_id
```
Comment (JordiMForgeFlow) [inline account_move_line_sale_info/hooks.py]:
```suggestion
        UPDATE account_move_line aml SET sale_line_id = sm.sale_line_id
```

File: account_move_line_sale_info/hooks.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        FROM account_move_line aml2
+        INNER JOIN stock_move sm ON
+        aml2.stock_move_id = sm.id
+        WHERE aml.id = aml2.id;
+    """
+    )
+    # FOR invoices
+    cr.execute(
+        """
+        update account_move_line aml set sale_line_id = sol.id
```
Comment (JordiMForgeFlow) [inline account_move_line_sale_info/hooks.py]:
```suggestion
        UPDATE account_move_line aml SET sale_line_id = sol.id
```

File: account_move_line_sale_info/hooks.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        FROM sale_order_line so
+        LEFT JOIN account_move_line aml2
+        ON aml2.sale_order_id = so.id
+        WHERE aml2.move_id = aml.move_id
+    """
+    )
+
+    cr.execute(
+        """
+        update account_move_line aml set sale_line_id = sol.id
```
Comment (JordiMForgeFlow) [inline account_move_line_sale_info/hooks.py:61]:
```suggestion
        UPDATE account_move_line aml SET sale_line_id = sol.id
```

File: account_move_line_sale_info/hooks.py (hunk 3 above)
Comment (AaronHForgeFlow) [reply account_move_line_sale_info/hooks.py]: Done. Thanks.

File: account_move_line_sale_info/hooks.py (hunk 4 above)
Comment (AaronHForgeFlow) [reply account_move_line_sale_info/hooks.py:61]: Done. Thanks.

File: account_move_line_sale_info/hooks.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    )
+
+    # NOW we can fill the lines without invoice_id (Odoo put it very
+    # complicated)
+
+    cr.execute(
+        """
+        UPDATE account_move_line aml
+        SET sale_order_id = so.id
+        FROM sale_order_line so
```
Comment (JordiMForgeFlow) [inline account_move_line_sale_info/hooks.py]:
```suggestion
        FROM sale_order so
```

Comment (AaronHForgeFlow) [reply account_move_line_sale_info/hooks.py]: Done!

**Comment by AaronHForgeFlow** [comment]
@MiquelRForgeFlow can you take a look to the migration scripts?

File: account_move_line_sale_info/tests/test_account_move_line_sale_info.py [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+from odoo.tests import common
+
+
+class TestAccountMoveLineSaleInfo(common.TransactionCase):
+    def setUp(self):
+        super(TestAccountMoveLineSaleInfo, self).setUp()
+        self.sale_model = self.env["sale.order"]
+        self.sale_line_model = self.env["sale.order.line"]
+        self.invoice_model = self.env["account.move"]
+        self.invoice_line_model = self.env["account.move.line"]
```
Comment (MiquelRForgeFlow) [inline account_move_line_sale_info/tests/test_account_move_line_sale_info.py]:
```suggestion
        self.am_model = self.env["account.move"]
```
line model is repeated

**Review by JordiBForgeFlow — APPROVED** [review]

**Pull request merged**